bluetooth/src/commonMain/kotlin/Bluetooth.kt

```diff
@@ -0,0 +1,296 @@
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+ */
+
+package com.splendo.kaluga.bluetooth
+
+import com.splendo.kaluga.bluetooth.device.BaseAdvertisementData
+import com.splendo.kaluga.bluetooth.device.ConnectionSettings
+import com.splendo.kaluga.bluetooth.device.Device
+import com.splendo.kaluga.bluetooth.device.DeviceAction
+import com.splendo.kaluga.bluetooth.device.DeviceInfoImpl
+import com.splendo.kaluga.bluetooth.device.DeviceState
+import com.splendo.kaluga.bluetooth.device.Identifier
+import com.splendo.kaluga.bluetooth.scanner.BaseScanner
+import com.splendo.kaluga.bluetooth.scanner.ScanningState
+import com.splendo.kaluga.bluetooth.scanner.ScanningStateRepo
+import com.splendo.kaluga.logging.d
+import com.splendo.kaluga.logging.info
+import com.splendo.kaluga.permissions.Permissions
+import kotlin.jvm.JvmName
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.MainScope
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.flow.combine
+import kotlinx.coroutines.flow.distinctUntilChanged
+import kotlinx.coroutines.flow.emptyFlow
+import kotlinx.coroutines.flow.first
+import kotlinx.coroutines.flow.flatMapLatest
+import kotlinx.coroutines.flow.flowOf
+import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.flow.mapLatest
+import kotlinx.coroutines.flow.stateIn
+import kotlinx.coroutines.flow.transformLatest
+import kotlinx.coroutines.launch
+
+class Bluetooth internal constructor(
+    permissions: Permissions,
+    connectionSettings: ConnectionSettings,
+    autoRequestPermission: Boolean,
+    autoEnableBluetooth: Boolean,
+    scannerBuilder: BaseScanner.Builder,
+    coroutineScope: CoroutineScope
+) : CoroutineScope by coroutineScope {
+
+    interface Builder {
+        fun create(
+            connectionSettings: ConnectionSettings = ConnectionSettings(ConnectionSettings.ReconnectionSettings.Always),
+            autoRequestPermission: Boolean = true,
+            autoEnableBluetooth: Boolean = true,
+            coroutineScope: CoroutineScope = MainScope()
+        ): Bluetooth
+    }
+
+    companion object {
+        private const val LOG_TAG = "Kaluga Bluetooth"
+    }
+
+    internal val scanningStateRepo = ScanningStateRepo(permissions, connectionSettings, autoRequestPermission, autoEnableBluetooth, scannerBuilder, coroutineScope)
+
+    sealed class ScanMode {
+        object Stopped:ScanMode()
+        class Scan(val filter: Set<UUID>):ScanMode()
+    }
+
+    private val scanMode = MutableStateFlow<ScanMode>(ScanMode.Stopped)
+
+    fun collectRepo(it:Any) {
```

Comment:
Function name doesn't reflect what it does

Reply:
It does (look at the callee), but it's just a debug function that should not be there

Reply:
since removed